### PR TITLE
Implement Configuration File Analyzer Worker (including 3 analyzers)

### PR DESCRIPTION
**Pull request by hacktobeer**
This PR implements a new worker capable of analyzing various configuration files, including SSH, Jupyter Notebook, and Jenkins, for security weaknesses. It includes the necessary analyzers, tasks, tests, and updates to the Dockerfile and documentation.

### Technical Details:
* **New Analyzers:**  Created new Python modules in `src/analyzers` to handle the analysis of different configuration file types:
    * `jenkins_analyzer.py`: Analyzes Jenkins `config.xml` files for weak credentials and extracts the Jenkins version.  Leverages Hashcat or John the Ripper to crack weak passwords.
    * `jupyter_analyzer.py`: Analyzes Jupyter Notebook configuration files for insecure settings such as disabled XSRF protection, root access, and remote access.
    * `sshd_analyzer.py`: Analyzes SSH daemon configuration files (`sshd_config`) for insecure settings such as `PermitRootLogin`, `PasswordAuthentication`, and `PermitEmptyPasswords`.
    * `utils.py` (moved to `src/analyzers/__init__.py`): Provides utility functions, including `bruteforce_password_hashes` which uses Hashcat or John for password cracking.
* **New Tasks:** Implemented Celery tasks in `src` to handle file processing and reporting for each analyzer:
    * `jenkins_task.py`: Processes files identified as `config.xml` (JenkinsConfigFile) and invokes the `jenkins_analyzer`.
    * `jupyter_task.py`: Processes files identified as `JupyterConfigFile` artifact and invokes the `jupyter_analyzer`.
    * `sshd_task.py` (renamed from `tasks.py`): Processes files identified as `SshConfigFile` artifact and invokes the `sshd_analyzer`.
* **Testing:** Added unit tests in `tests` to ensure the functionality of each analyzer:
    * `test_jenkins_analyzer.py`: Tests version and credential extraction, as well as the analysis logic, mocking the `bruteforce_password_hashes` function.
    * `test_jupyter_analyzer.py`: Tests the analysis logic for various Jupyter Notebook misconfigurations.
    * `test_sshd_analyzer.py`: Tests the analysis logic for both empty and weak SSH configurations.
* **Dockerfile Updates:** Modified the `Dockerfile` to include the required packages for password cracking (Hashcat, John the Ripper, John data, pocl-opencl-icd, and curl to download password lists) and to specify the platform as `linux/amd64`. Added logic to download and prepare password lists.
* **Dependency Updates:** Updated dependencies in `pyproject.toml` and `poetry.lock` to include testing libraries (`pytest`, `pytest-cov`) and other required packages (`debugpy`, `packaging`, `pluggy`, `tomli`). Updated the `openrelik-worker-common` package.
* **Documentation:** Updated the `README.md` file with instructions for deploying the new worker in a Docker Compose environment. It details supported configuration analyzers and provides an example configuration snippet.
* **Application Updates:** Updated the `app.py` to include the new tasks in the Celery app.

**Note: Analyzers have been ported from Turbinia.**

Screenshot 2024-10-11 at 12 06 03

### CODE REVIEW

**Comment by hacktobeer** [comment]
@berggren pytests will pass if you push a new release of the openrelik-common-worker library which include the reporting helper functions.

**Comment by berggren** [comment]
> @berggren pytests will pass if you push a new release of the openrelik-common-worker library which include the reporting helper functions.

Ack

**Comment by berggren** [comment]
File structure suggestion:

The naming is a bit redundant IMO. To follow what we do in other workers, mayabe something like this:
```
src
├── __init__.py
├── analyzers
│   ├── __init__.py
│   ├── jenkins_analyzer.py
│   ├── jupyter_analyzer.py
│   ├── sshd_analyzer.py
│   └── utils.py
├── app.py
├── jenkins_task.py
├── jupyter_task.py
└── sshd_task.py
```

**Comment by hacktobeer** [comment]
@berggren files renamed and import in app.py reflect that.

**Comment by berggren** [comment]
@hacktobeer common lib has been updated with the new reporting markdown feature. Do you want to update this PR to reflect that?

File: src/jenkins_task.py [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    Returns:
+        Base64-encoded dictionary containing task results.
+    """
+    input_files = get_input_files(pipe_result, input_files or [])
+    output_files = []
+
+    for input_file in input_files:
+        if (
+            input_file.get("data_type").lower()
+            == f"openrelik.worker.file.{EXPECTED_FILENAME}".lower()
```
Comment (berggren) [inline src/jenkins_task.py:66]: This is going to change soon (the data_type) but we can adjust that afterwards

File: src/jenkins_task.py [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+        ):
+            output_file = create_output_file(
+                output_path,
+                filename=f"{input_file.get('filename')}-{SHORT_TASK_NAME}-report",
```
Comment (berggren) [inline src/jenkins_task.py:70]: Needs updating to the new create_output_file() implementation

filename --> display_name

**Review by berggren — CHANGES_REQUESTED** [review]
Overall looks good. A couple of comments and I'll take another look when the reporting has been updated to the new implementation

**Comment by hacktobeer** [comment]
@berggren Refactored for the new reporter class, PTAL.

**Review by berggren — APPROVED** [review]

**Pull request merged**